File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    heapq.heapify(scoville)
    cnt = 0
    while scoville:
        a = heapq.heappop(scoville)
        if a >= K:
            return cnt
        
        if not scoville:
            return -1
        
        b = heapq.heappop(scoville)
        c = a + 2 * b
        heapq.heappush(scoville, c)
        
        cnt += 1
        
    return -1
        

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.PriorityQueue;

class Solution {
    public int solution(int[] scoville, int K) {
        PriorityQueue<Integer> heap = new PriorityQueue<>();
        int cnt = 0;
        
        for (int s : scoville) {
            heap.add(s);
        }
        
        while (true) {
            int a = heap.poll();
            
            if (a >= K) {
                break;  
            }
            

            if (heap.isEmpty()) {
                return -1;
            }
            
            int b = heap.poll();
            int c = a + b * 2;
   
            heap.add(c);
            cnt += 1;
        }
        
        return cnt;
    }
}